extensions/amp-analytics/0.1/linker.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {crc32} from './crc32';
+
+/** @const {string} */
+const DELIMITER = '~';
+
+/**
+ * Creates a new query string parameter to be used for cross-domain ID syncing.
+ * &<paramName>=<version>~<checksum>~<key1>~<value1>~<key2>~<value2>...
+ */
+export class Linker {
+  /**
+   * Creates the linker param from the given config, and returns the url with
+   * the given param attached.
+   * @param {string} version
+   * @param {Object<string,string>} pairs
+   * @return {string}
+   */
+  create(version, pairs) {
+    if (!pairs || !Object.keys(pairs).length) {
+      return '';
+    }
+
+    return this.generateParam_(version, pairs);
+  }
+
+
+  /**
+   * Generate the completed querystring.
+   * <paramName>=<version>~<checksum>~<key1>~<value1>~<key2>~<value2>...
+   * @param {string} version
+   * @param {Object<string, string>} pairs
+   * @return {string}
+   */
+  generateParam_(version, pairs) {
+    const encodedPairs = this.encodePairs_(pairs);
+    const checksum = this.getCheckSum_(encodedPairs);
+    return version + DELIMITER + checksum + encodedPairs;
```

Comment:
missing DELIMITER in between checksum & encodedPairs

Reply:
Talked offline. Changed this logic around to be easier to read.